galley/pkg/config/analysis/analyzers/testdata/incorrect-port-name-external-name-service-type.yaml

```diff
@@ -0,0 +1,28 @@
+# If port is unnamed or port name doesn't follow <protocol>[-<suffix>], when working with ExternalName service type, the analyzer will report an Error.
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: nginx
+  namespace: nginx-ns
+spec:
+  externalName: nginx.example.com
+  ports:
+  - name: nginx
+    port: 443
+    protocol: TCP
```

Comment:
Note protocol here is declared, it is TCP. [-suffix] is optional when declaring protocol.

cc @bianpengyuan for input also